marketplace/neo4j.template.yaml

```diff
@@ -424,15 +424,27 @@ Resources:
             - "install_neo4j_from_yum() {\n"
             - "  echo \"Installing Graph Database...\"\n"
             - "  export NEO4J_ACCEPT_LICENSE_AGREEMENT=yes\n"
-            - "  if ! curl --fail http://versions.neo4j-templates.com/target.json; then\n"
+            - "  ASG_NAME=$(aws --region $region cloudformation describe-stack-resources --stack-name $stackName --query 'StackResources[?ResourceType==`AWS::AutoScaling::AutoScalingGroup`].PhysicalResourceId' --output text)\n"
+            - "  NEO4J_VERSION=$(aws --region $region autoscaling describe-auto-scaling-groups --auto-scaling-group-names $ASG_NAME --query 'AutoScalingGroups[0].Tags[?Key==`NEO4J_VERSION`].Value' --output text)\n"
+            - "  if [[ ! -z $NEO4J_VERSION ]]; then\n"
+            - "    echo \"Found NEO4J_VERSION Tag on AutoScalingGroup: NEO4J_VERSION=$NEO4J_VERSION\"\n"
+            - "    local -r graphDatabaseVersion=\"neo4j-enterprise-$NEO4J_VERSION\"\n"
+            - "  elif ! curl --fail http://versions.neo4j-templates.com/target.json; then\n"
```

Comment:
this condition is pretty complex to understand to me...! curl --fail ...does this mean the else if condition is satisfied if the curl fails...
my suggestion would be to do curl only once outside the if and store the output and based on the output manage the elif and else ...this can help improve readability a bit more

Reply:
Code updated as discussed. 

I have also added conditional logic to only tag the ASG if it is not already tagged